### PR TITLE
Dailymotion Bid Adapter: add consent enforcement to read the advertising cookie

### PR DESCRIPTION
**Pull request by kvnsw**
## Type of change
* [x]  Feature

## Description of change
* Add consent enforcement before reading the advertising cookie
* If Dailymotion does not have consent from the user, it does not transmit any cookie in the request to the Prebid server (previously the cookie was sent but not used)

### CODE REVIEW

File: modules/dailymotionBidAdapter.js [hunk 1]
```diff
@@ -130,76 +130,96 @@ export const spec = {
...
+          auctionId: deepAccess(bid, 'auctionId', ''),
+          bidId: deepAccess(bid, 'bidId', ''),
+          mediaTypes: {
+            video: {
+              api: bid.mediaTypes?.[VIDEO]?.api || [],
+              mimes: bid.mediaTypes?.[VIDEO]?.mimes || [],
+              minduration: bid.mediaTypes?.[VIDEO]?.minduration || 0,
+              maxduration: bid.mediaTypes?.[VIDEO]?.maxduration || 0,
+              playbackmethod: bid.mediaTypes?.[VIDEO]?.playbackmethod || [],
+              plcmt: bid.mediaTypes?.[VIDEO]?.plcmt || 1, // Fallback to instream considering logic of `isBidRequestValid`
```
Comment (patmmccann) [inline modules/dailymotionBidAdapter.js:203]: It is not appropriate to fill this in with a fallback to instream; instream must be declared by the publisher. You're putting their inventory at risk for getting banned by DSPs for incorrect declarations.

File: modules/dailymotionBidAdapter.js [hunk 2]
```diff
@@ -130,76 +130,96 @@ export const spec = {
...
+              protocols: bid.mediaTypes?.[VIDEO]?.protocols || [],
+              skip: bid.mediaTypes?.[VIDEO]?.skip || 0,
+              skipafter: bid.mediaTypes?.[VIDEO]?.skipafter || 0,
+              skipmin: bid.mediaTypes?.[VIDEO]?.skipmin || 0,
+              startdelay: bid.mediaTypes?.[VIDEO]?.startdelay || 0,
```
Comment (patmmccann) [inline modules/dailymotionBidAdapter.js:208]: It is not appropriate to fill this in with a fallback; preroll must be declared by the publisher. You're putting their inventory at risk for getting banned by DSPs for incorrect declarations.

File: modules/dailymotionBidAdapter.js [hunk 3]
```diff
@@ -130,76 +130,96 @@ export const spec = {
...
+  buildRequests: function(validBidRequests = [], bidderRequest) {
+    // check consent to be able to read user cookie
+    const allowCookieReading =
+      // No GDPR applies
+      !deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ||
+      // OR GDPR applies and we have global consent
+      deepAccess(bidderRequest, 'gdprConsent.vendorData.hasGlobalConsent') === true ||
+      (
+        deepAccess(bidderRequest, 'gdprConsent.vendorData.vendor.consents.573') === true &&
+        deepAccess(bidderRequest, 'gdprConsent.vendorData.purpose.consents.1') === true &&
```
Comment (patmmccann) [inline modules/dailymotionBidAdapter.js:142]: surely there is a more compact way to code this

**Review by patmmccann — CHANGES_REQUESTED** [review]
a few minor changes

File: modules/dailymotionBidAdapter.js (hunk 3 above)
Comment (patmmccann) [reply modules/dailymotionBidAdapter.js:142]: maybe something like this

`const li = 'legitimateInterests';
const c = 'consents';

const hasConsent = (type, path) => deepAccess(bidderRequest.gdprConsent.vendorData.purpose[type], path) === true;

const allowCookieReading = 
  // No GDPR applies
  !deepAccess(bidderRequest, 'gdprConsent.gdprApplies') ||
  // OR GDPR applies and we have global consent
  deepAccess(bidderRequest, 'gdprConsent.vendorData.hasGlobalConsent') === true ||
  // OR specific consents are given
  (
    deepAccess(bidderRequest, 'gdprConsent.vendorData.vendor.consents.573') === true &&
    hasConsent(c, '1') &&
    hasConsent(c, '3') &&
    hasConsent(c, '4') &&
    // emulate flexible purpose by checking if the default consent or legitimate interest is set
    (hasConsent(c, '2') || hasConsent(li, '2')) &&
    (hasConsent(c, '7') || hasConsent(li, '7')) &&
    (hasConsent(c, '9') || hasConsent(li, '9')) &&
    (hasConsent(c, '10') || hasConsent(li, '10'))
  );`

Comment (kvnsw) [reply modules/dailymotionBidAdapter.js]: Done! Let me know what you think about this new proposal

**Review by patmmccann — APPROVED** [review]

**Pull request merged**